MapboxCoreNavigation/CLLocation.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import CoreLocation
+import MapboxDirections
+
+public extension CLLocation {
+    /**
+     Returns a `CLLocation` infront of the provided location given the interval
+     with the current speed taken into account.
+     */
+    public func advance(by interval: TimeInterval) -> CLLocation {
+        let metersAhead = speed * interval
+        let locationInfrontOfUser = coordinate.coordinate(at: metersAhead, facing: course)
+        return CLLocation(coordinate: locationInfrontOfUser.coordinate(at: metersAhead, facing: course),
+                          altitude: altitude,
+                          horizontalAccuracy: horizontalAccuracy,
+                          verticalAccuracy: verticalAccuracy,
+                          course: course,
+                          speed: speed,
+                          timestamp: timestamp.addingTimeInterval(interval))
+    }
+    
+    public func isOnRoute(_ routeProgress: RouteProgress) -> Bool {
```

Comment:
`_` is only appropriate if `routeProgress` is a “route”. In this case, since `routeProgress` is a “route progress”, call this method `isOnRoute(with:)` or have it take the current and upcoming steps directly: `isOn(_:upComingStep:)`.